pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,585 @@
+    PolicyFeatureSpec policy = 26;
+    TelemetryFeatureSpec telemetry = 27;
+    SecurityFeatureSpec security = 28;
+    ConfigManagementFeatureSpec config_management = 29;
+
+    // Ingress/egress gateway installation and configuration.
+    repeated IngressGatewaySpec ingress_gateway = 30;
+    repeated EgressGatewaySpec egress_gateway = 31;
+
+    // List of external operators that will be installed (but not managed)
+    // by the IstioInstaller e.g. prometheus, grafana, kiali.
+    repeated OperatorSpec external_operators = 50;
+
+    // Global k8s defaults. Can be overridden per component.
+    KubernetesResourcesSpec k8s_defaults = 80;
+
+    // URL for the base spec. e.g.
+    //   istio-auth.yaml (looks in default dir)
+    //   file://var/istio/install/values/custom/custom-install.yaml
+    // Uses compiled in default install spec by default.
+    string base_spec_path = 100;
+    // URL for the install package. e.g.
+    //   http://github.com/istio/istio/releases/install/1.1.2
+    //   http://github.com/istio/istio/releases/install/lts
+    //   file://tmp/istio-installer/nightly
+    // Uses compiled in install package by default.
+    string custom_package_path = 102;
+    // Root for docker image paths e.g. docker.io/istio-release.
+    // Releases are published to docker hub under 'istio' project.
+    // Daily builds from prow are on gcr.io, and nightly builds from circle on docker.io/istionightly
+    string hub = 103;
+    // Version tag for docker images e.g. 1.0.6
+    string tag = 104;
+}
+
+// Configuration options for traffic management.
+message TrafficManagementFeatureSpec {
+    // Selects whether traffic management is installed.
+    // Must be enabled to enable any sub-component.
+    google.protobuf.BoolValue enabled = 1;
+    // Namespace that all traffic management components are installed into.
+    string namespace = 2;
+
+    // Component specific config.
+    PilotComponentSpec pilot = 30;
+    ProxyComponentSpec proxy = 31;
+    SidecarInjectorComponentSpec sidecar_injector = 32;
+}
+
+// Configuration options for the policy feature.
+message PolicyFeatureSpec {
+    // Selects whether policy is installed.
+    // Must be enabled to enable any sub-component.
+    google.protobuf.BoolValue enabled = 1;
+    // Namespace that all policy components are installed into.
+    string namespace = 2;
+
+    // Default behavior of the sidecar for handling outbound traffic from the application.
+    enum OutboundPolicy {
+        // Outbound traffic to unknown destinations will be allowed, in case there are no
+        // services or ServiceEntries for the destination port
+        ALLOW_ANY = 0;
+        // Restrict outbound traffic to services defined in the service registry as well
+        // as those defined through ServiceEntries
+        REGISTRY_ONLY = 1;
+    }
+
+    // If set, allows traffic in cases when the mixer policy service cannot be reached.
+    google.protobuf.BoolValue policy_check_fail_open = 10;
```

Comment:
can user also configure disable policy check? that seems the most important config for mixer policy.

Reply:
for example, in 1.1, we have mixer policy enabled by default but disable policy check enabled by default.

Reply:
that part is a bit confusing for me. how is installing policy but disabling the check different from not installing policy at all and why do we want to support this scenario?

Reply:
ok, these are 2 different install configs.  In 1.1, we wanted to not install policy and disable policy check but we ended up shipping disable policy check but having policy installed by default due to backwards compatible with 1.0.  Installing policy maps to have mixer policy deployment and services installed, disable policy check is a config we expose to user when mixer is installed where user can choose to skip all policy checks.

Reply:
That is up to the mixer team and backwards compatible when moving from 1.1.

If mixer v2 lands in 1.3, then we won't need this for sure.